compiler/include/symbol.h

```diff
@@ -361,6 +379,45 @@ class ArgSymbol : public LcnSymbol {
 *                                                                   *
 ********************************* | ********************************/
 
+class ShadowVarSymbol : public VarSymbol {
+public:
+  ShadowVarSymbol(ForallIntentTag iIntent, const char* iName, Expr* iSpec = NULL);
+
+  virtual void    verify();
+  virtual void    accept(AstVisitor* visitor);
+  DECLARE_SYMBOL_COPY(ShadowVarSymbol);
+
+  virtual void    replaceChild(BaseAST* oldAst, BaseAST* newAst);
+  virtual bool    isConstant()                              const;
+  virtual bool    isConstValWillNotChange()                 const;
+
+  const char* intentDescrString() const;
+  // The corresponding outer var or NULL if not applicable.
+  SymExpr* outerVarSE() const { return (SymExpr*)outerVarRep; }
+  bool     isReduce()   const { return intent == TFI_REDUCE;  }
+  Expr*    spec()       const;
+  void     removeSupportingReferences();
+
+  ForallIntentTag intent;
+
+  // This would be a SymExpr*, if not for checkIdInsideWithClause().
+  // See also: sv->outerVarSE() and outerVarSym(sv).
+  Expr* outerVarRep;
+
+  // For a reduce intent, the reduce expression.
+  // For a task-private variable, the initialization expression.
+  // Either way, wrapped in a block.  Otherwise it is NULL.
+  BlockStmt* specBlock;
+
+  Symbol* reduceGVar;
```

Comment:
I don't know what reduceGVar is. Is that the "global" variable? Maybe name it reduceTotal or something like that?

Reply:
Given that "globalOp" is a name that has been used throughout our implementation of reduce intents, historically and at present, how about reduceGVar --> reduceGlobalOp ?

Reply:
I don't have a problem having a field with the name reduceGlobalOp, but it would seem to me that such a field would store something different from a field named reduceGlobalVar. (The reduceGlobalOp would store a reduction class instance or "operator"; but a reduceGlobalVar might store the result of the reduction, say).

Reply:
Since it is indeed a reduction class instance, not the reduction state or final result, I will go with reduceGlobalOp.